pkcs11sec/pkcs11_security.go

```diff
@@ -0,0 +1,642 @@
+package pkcs11sec
+
+import (
+	"bytes"
+	"context"
+	"crypto"
+	"crypto/rsa"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"github.com/choria-io/go-security/filesec"
+	"github.com/howeyc/gopass"
+	"github.com/miekg/pkcs11"
+	"github.com/miekg/pkcs11/p11"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"path/filepath"
+	"regexp"
+	"sort"
+	"strings"
+	"time"
+)
+
+var hashPrefixes = map[crypto.Hash][]byte{
```

Comment:
might be worth adding a comment here to tell us where to get these from for when new standards come around?